packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx

```diff
@@ -0,0 +1,442 @@
+      return header.clientHeight;
+    }
+    return 0;
+  }
+
+  // add margin to the list based off the header element height
+  private addListMargin(): void {
+    const content = this.el.shadowRoot?.querySelector(
+      `.${TOC_CONTENT_CLASS}`
+    ) as HTMLDivElement;
+    content?.style.setProperty("top", `${this.headerHeight}px`);
+  }
+
+  // add tooltip to the item if it is too wide
+  private addTooltip(item: any, widthBreakpoint: number) {
+    const itemLabel = item.querySelector(".table-of-contents-item-label");
+    if (item.parentNode?.tagName == "IC-TOOLTIP") {
+      const tooltip = item.parentNode;
+      if (itemLabel?.offsetWidth < itemLabel?.scrollWidth) {
+        const tooltipPosition =
+          window.innerWidth < widthBreakpoint
+            ? "bottom"
+            : this.position === "right"
+            ? "left"
+            : "right";
+        //Modify the ic-tooltip wrapped around the list item
+        tooltip.setAttribute("disable-click", false);
+        tooltip.setAttribute("disable-hover", false);
+        tooltip.setAttribute("placement", tooltipPosition);
+        tooltip.setAttribute("target", item.id);
+        tooltip.setAttribute("style", "overflow:hidden");
+      } else {
+        //disable the tooltip if the item does not need to truncate
+        tooltip.setAttribute("disable-click", true);
+        tooltip.setAttribute("disable-hover", true);
+      }
+    }
+  }
+
+  // truncate text if it goes out of bounds of the container
+  private truncateText = () => {
+    const links = this.el.shadowRoot?.querySelectorAll(`.${TOC_ITEM_CLASS}`);
+    const contentList = this.el.shadowRoot?.querySelector(
+      `.${TOC_LIST_ITEMS_CLASS}`
+    );
+    if (contentList)
+      links?.forEach((item) => {
+        this.addTooltip(item, DEVICE_SIZES.S);
+      });
+  };
+
+  // initialise resize observer
+  componentWillLoad(): void {
+    this.resizeObserver = new ResizeObserver(() => this.handleResizeBounds());
+    this.resizeObserver.observe(this.el);
+  }
+
+  // initialise scroll listener
+  componentDidRender(): void {
+    document.addEventListener("scroll", () => {
+      if (!this.suppressListener) this.updateSelectedItemFromScroll();
+      else this.suppressListener = false;
+    });
+    this.containerPosition();
+  }
+
+  // after component has loaded modify the list based of position, size and surrounding elements
+  async componentDidLoad(): Promise<void> {
+    this.focusedElement = 0;
+    this.headerHeight = this.findHeaderElement();
+    this.addListMargin();
+    this.truncateText();
+    this.findElementsHeights();
+  }
+
+  // Component is a wrapper for page contents, and will build list from heading items
+  render() {
+    const { heading, monochrome, position, size, theme } = this;
+    return (
+      <Host class={{ [`ic-theme-${theme}`]: theme !== "inherit" }}>
+        <ic-section-container
+          id={"table-of-contents-section-container"}
+          aligned={"full-width"}
```

Comment:
I think these can just be:
```suggestion
          id="table-of-contents-section-container"
          aligned="full-width"
```